### PR TITLE
Update component to NPM + Flow 2.0

### PR DESCRIPTION
**Pull request by caalador**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-menu-bar-flow/21)

### CODE REVIEW

File: vaadin-menu-bar-flow/pom.xml [hunk 1]
```diff
@@ -27,11 +27,6 @@
             <artifactId>vaadin-button</artifactId>
             <version>2.1.4</version>
         </dependency>
-        <dependency>
-            <groupId>org.webjars.bowergithub.vaadin</groupId>
-            <artifactId>vaadin-context-menu</artifactId>
```
Comment (pekam) [inline vaadin-menu-bar-flow/pom.xml]: Why is this removed? Doesn't it break the "Fail if non-pinned dependencies found"-check in snapshot build?

File: vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java [hunk 2]
```diff
@@ -40,6 +42,8 @@
  */
 @Tag("vaadin-menu-bar")
 @HtmlImport("frontend://bower_components/vaadin-menu-bar/src/vaadin-menu-bar.html")
+@JsModule("@vaadin/vaadin-menu-bar/vaadin-menu-bar.js")
+@NpmPackage(value = "@vaadin/vaadin-menu-bar", version = "1.0.0-alpha4")
```
Comment (pekam) [inline vaadin-menu-bar-flow/src/main/java/com/vaadin/flow/component/menubar/MenuBar.java]: Bump to beta1 after updating the changes from master.

File: vaadin-menu-bar-flow/pom.xml (hunk 1 above)
Comment (caalador) [reply vaadin-menu-bar-flow/pom.xml]: vaadin-context-menu-flow defines the webjars version of vaadin-context-menu. Why would we want to override that version with a perhaps wrong version of the webjar?

Comment (pekam) [reply vaadin-menu-bar-flow/pom.xml]: Because of the requirement to pin all webjar dependencies. I tested locally the "Fail if non-pinned dependencies found"-script and it fails without this dep. So the snapshot build will keep failing as long as we have that check in place.

Comment (caalador) [reply vaadin-menu-bar-flow/pom.xml]: Can't see how it can be seen a unpinned as it is pinned in the dependency... Also where is this magic script located?

Comment (pekam) [reply vaadin-menu-bar-flow/pom.xml]: It's [here](https://bender.vaadin.com/admin/editRunType.html?id=buildType:Flow_ComponentSnapshot_MenuBar&runnerId=RUNNER_287&cameFromUrl=%2Fadmin%2FeditBuildRunners.html%3Fid%3DbuildType%253AFlow_ComponentSnapshot_MenuBar%26init%3D1&cameFromTitle=), the second build step in all component snapshot builds.

**Comment by manolo** [comment]
TC Failure is because some wrong dependency.
There is no reviewable here, shouldn't be configured?

**Comment by pekam** [comment]
@manolo, the components team has switched to using GitHub reviews instead of Reviewable, which is why Reviewable is not configured for new projects.

**Comment by caalador** [comment]
@manolo the bower versions aren't really linked to a npm update.

**Comment by tomivirkki** [comment]
Is this still blocked by something?

**Comment by caalador** [comment]
manolo requested changes
Else no.
